### ISSUE TITLE
Multi-level item in `selectionSet` is not received.

### ISSUE THREAD

**Issue by erickriva** [aws-amplify/amplify-js#12665]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

GraphQL API

### Environment information

<details>

```
  System:
    OS: Windows 11 10.0.23590
    CPU: (16) x64 AMD Ryzen 9 5900HX with Radeon Graphics
    Memory: 13.21 GB / 23.91 GB
  Binaries:
    Node: 18.18.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.8.1 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.11.0 - ~\AppData\Local\pnpm\pnpm.EXE
    bun: Not Found
    Watchman: Not Found
  npmPackages:
    @aws-amplify/backend: 0.5.5
    @aws-amplify/backend-cli: 0.9.1
    aws-amplify: 6.0.5
    aws-cdk: Not Found
    aws-cdk-lib: 2.111.0
    typescript: 5.3.2
```

</details>

### Describe the bug

A multi-level item in `selectionSet` is not received.

### Expected behavior

Expected to receive data for any level of nested query.

### Reproduction steps

- Declare schema as following:
```javascript
import { a } from "@aws-amplify/backend";

const schema = a.schema({
  CommunityPost: a.model({
    id: a.id().required(),
    poll: a.hasOne("CommunityPoll"),
  }),

  CommunityPoll: a.model({
    id: a.id().required(),
    question: a.string().required(),
    answers: a.hasMany("CommunityPollAnswer").arrayRequired().valueRequired(),
  }),

  CommunityPollAnswer: a.model({
    id: a.id().required(),
    answer: a.string().required(),
    votes: a.hasMany("CommunityPollVote").arrayRequired().valueRequired(),
  }),

  CommunityPollVote: a.model({ id: a.id().required() }),
});
```

- Try to get data using:
```javascript
amplifyClient.models.CommunityPost.get(
    { id: 'post id' },
    {
      selectionSet: [
        "id",
        "poll.id",
        "poll.question",
        "poll.answers.id",
        "poll.answers.answer",
        "poll.answers.votes.id",
      ],
    }
  );
```

- `CommunityPost` is received.
 - `poll` properties are received.
 - `poll.answers` properties are received, except for `poll.answers.votes`

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by erickriva** [comment]
Testing directly through AppSync, with query:

```graphql
query MyQuery {
  listCommunityPosts {
    items {
      id
      poll {
        question
        answers {
          items {
            answer
            votes {
              items {
                id
              }
            }
          }
        }
      }
    }
  }
}
```

Following data is returned:
```
{
  "data": {
    "listCommunityPosts": {
      "items": [
        {
          "id": "2",
          "poll": null
        },
        {
          "id": "1",
          "poll": {
            "question": "Is this a good question for showing in issue?",
            "answers": {
              "items": [
                { "answer": "Yes", "votes": { "items": [] } },
                { "answer": "No",  "votes": { "items": [ { "id": "f0396484-e3c8-4fbc-87cb-1811bad7b4af" } ]  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}
```

**Comment by erickriva** [comment]
I think I may have confused up things (sorry, it was almost midnight 🥱), but seems there's a bug here still.

Request:
```json
{
  "query": "query ($id: ID!) {\n  getCommunityPost(id: $id) {\n    id\n      poll {\n      id\n      question\n      answers {\n        items {\n          votes {\n            items {\n              id\n            }\n          }\n        }\n      }\n    }\n  }\n}\n",
  "variables": {
    "id": "1"
  }
}
```

Response:
```json
{
  "data": {
    "getCommunityPost": {
      "id": "1",
      "poll": {
        "id": "1",
        "question": "Is this a good question for showing in issue?",
        "answers": {
          "items": [
            {
              "votes": {
                "items": []
              }
            },
            {
              "votes": {
                "items": [
                  {
                    "id": "f0396484-e3c8-4fbc-87cb-1811bad7b4af"
                  }
                ]
              }
            }
          ]
        }
      }
    }
  }
}
```

They seems to match between them, but not with my code: `selectionSet: [ "id", "poll.id", "poll.question", "poll.answers.id", "poll.answers.answer", "poll.answers.votes.id" ]`

`answers.id` and `answers.answer` aren't present in the request, so no response too. But `answers.votes.id` is present.

**Comment by erickriva** [comment]
@chrisbonifacio I've been trying to modify the lib's code for the last days and I think I have solved the issue. Don't know about performance yet. I will open a PR soon.

**Comment by erickriva** [comment]
In the case this PR is approved, I'll send another to fix [this issue](https://github.com/aws-amplify/amplify-category-api/issues/2096), at least the execution part of it. Maybe I can fix TS too.

**Comment by chrisbonifacio** [comment]
Hey @erickriva thank you for looking into this and submitting a PR! I will bring it to the team's attention.

**Comment by erickriva** [comment]
PR #12701 was merged, released in 6.0.9, and this problem was solved.
Thank you @chrisbonifacio!

**Comment by chrisbonifacio** [comment]
@erickriva thank YOU, sir! Excellent work 🙏
